server.js

```diff
@@ -0,0 +1,67 @@
+const express = require('express')
+const app = express()
+const port = 5000
+
+app.listen(port, () => {
+    console.log(`Started on port ${port}`);
+})
+
+const {getAllUsers, getUser, addUser, updateUser, deleteUser} = require('./controllers/usersController')
+
+//middleware
+app.use(express.json())
+
+const hello = (req, res) => {
+    res.send({ welcome: "Welcome to user DB"})
+}
+
+//Routes
+//home page
+app.get('/', hello)
+
+//all users
+app.get('/users', getAllUsers)
+
+//sign up a new user
+app.post('/users',addUser)
+
+//a single user by id
+app.get('/users/:id',getUser)
+
+//update existing user by id
+app.patch('/users/:id', updateUser)
+
+//delete existing user by id
+app.delete('/users/:id',deleteUser)
+
+
+
+//ex 2
+let users = [ 
+    {username: "User1", password: "PW1"}, 
+    {username: "User2", password: "PW2"}, 
+    {username: "User3", password: "PW3"}
+]
+/* app.post('/users', (req,res) => {
+    console.log("post users route called");
+    //data from frontend
+    console.log(req.body);
+    res.send({
+        message: 'User logged in successfully...'
+    })
+}) */
+
+app.post('/login', (req, res) => {
```

Comment:
Almost perfect until here :) 
The login route callback I would also put into the usersController file. Currently you now handle users from two different sources: You have one set of users in the hardcoded users array above and then another version of users that you manage in lowdb in the usersController. 

I would unify these two sets of users and comment out the hardcoded users array.

By putting the login route callback into the usersController, you can also use the users stored lowdb in your login function.

Then you will have a complete API where new, signed up users that you stored in lowdb will also be able to login.

Reply:
Thank you for feedback. 
Made changes. Please, check. Thank you.

Reply:
Wow, now it is perfect. Very well done!